src/main/fc/stats.c

```diff
@@ -91,6 +93,10 @@ void statsOnDisarm(void)
             statsConfigMutable()->stats_total_flights += 1;    // arm / flight counter
             statsConfigMutable()->stats_total_time_s += dtS;
             statsConfigMutable()->stats_total_dist_m += (DISTANCE_FLOWN_CM - arm_distance_cm) / 100;
+                if (statsNotSavedYet) {
```

Comment:
Add `&& getBatteryState() != BATTERY_NOT_PRESENT` or `&& getBatteryCellCount() > 0` here.

Reply:
Hi ! Done (locally, I'll commit it with the others changes).
I'm just wondering why this is needed ? My understanding the other stats above won't update with simple USB plugging, since the the stats are written only after disarm (so, after arm) and successful flight (_stats_min_armed_time_s_), and so will _stats_total_packs._ Will they ? Otherwise, that `&& getBatteryState() != BATTERY_NOT_PRESENT`, that indeed would be needed, would better be set alongside

```c
line 90.      if (minArmedTimeS >= 0) {
```

or

```c
line 105.     if (saveRequired) {
```

Wouldn't it ?

Reply:
What about the boards without voltmeters? Or there are no any?
Also i think `getBatteryCellCount()` is not a correct function as it returns a constant if force_battery_cell_count is set regardless wether the battery is plugged or not.

Reply:
yeah, if `getBatteryCellCount` has a default then it's probably not suitable.  Checking the battery state seems more appropriate.  i'm fairly sure all boards have voltmeters since the MCU is usually fed a voltage on an ADC pin via a resistor divider, probably a good idea to check this assumption though.